ansible_builder/main.py

```diff
@@ -59,6 +64,63 @@ def __init__(self,
             galaxy_ignore_signature_status_codes=galaxy_ignore_signature_status_codes)
         self.verbosity = verbosity
+        self.container_policy, self.container_keyring = self._handle_image_validation_opts(container_policy, container_keyring)
+
+    def _handle_image_validation_opts(self, policy, keyring):
+        """
+        Process the container_policy and container_keyring arguments.
+
+        :param str policy: The container_policy value.
+        :param str keyring: The container_keyring value.
+
+        The container_policy and container_keyring arguments come from the CLI
+        and work together to help build or use a podman policy.json file used
+        do image validation. Depending on the policy being used, the keyring
+        may or may not be necessary.
+
+        The keyring, if required, must be a valid path, and will be transformed
+        to an absolute path to be used in the policy.json file.
+
+        :returns: A tuple of a PolicyChoices enum and abs path to the keyring.
+        """
+        resolved_policy = None
+        resolved_keyring = None
+
+        if policy is not None:
+            if self.version != "2":
+                raise ValueError(f'--container-policy not valid with version {self.version} format')
+
+            # Require podman runtime
+            if self.container_runtime != 'podman':
+                raise ValueError('--container-policy is only valid with the podman runtime')
+
+            resolved_policy = PolicyChoices(policy)
+
+            # Require keyring if we write a policy file
+            if resolved_policy == PolicyChoices.SIG_REQ and keyring is None:
+                raise ValueError(f'--container-policy={resolved_policy.value} requires --container-keyring')
+
+            # Do not allow images to be defined with --build-arg CLI option if
+            # any sig policy is defined.
+            for key, _ in self.build_args.items():
+                if key in ('EE_BASE_IMAGE', 'EE_BUILDER_IMAGE'):
+                    raise ValueError(f'{key} not allowed in --build-arg option with --container-policy={resolved_policy.value}')
+
+        if keyring is not None:
+            # Require the correct policy to be specified
+            if resolved_policy is None:
+                raise ValueError('--container-keyring requires --container-policy')
+            elif resolved_policy != PolicyChoices.SIG_REQ:
+                raise ValueError(f'--container-keyring is not valid with --container-policy={resolved_policy.value}')
+
+            # Set the keyring to an absolute path to be referenced in the policy file.
+            if not os.path.exists(keyring):
+                raise ValueError('--container-keyring error: file does not exist')
+            if not os.path.isfile(keyring):
+                raise ValueError('--container-keyring error: not a file')
+            resolved_keyring = os.path.abspath(keyring)
```

Comment:
```suggestion
            resolved_keyring = os.path.abspath(keyring)
            logger.debug('Using Container Keyring from file %s', resolved_keyring)
```

Useful, not required.

Reply:
This is already logged in `build_command()` a bit further down in this file, line 196.